apps/ui/pages/reservation-unit/[id].tsx

```diff
@@ -377,54 +413,143 @@ const ReservationUnit = ({
+    [formUIDate]
+  );
 
-  useEffect(() => {
-    const scrollToCalendar = () => {
-      if (calendarRef?.current?.parentElement?.offsetTop != null) {
-        window.scroll({
-          top: calendarRef.current.parentElement.offsetTop - 20,
-          behavior: "smooth",
-        });
-      }
+  const timeValue = watch("time") ?? getTimeString();
+  const submitReservation = (_data: PendingReservationFormType) => {
+    if (focusSlot.start && focusSlot.end && reservationUnit.pk)
+      setErrorMsg(null);
+    const { start: begin, end } = focusSlot;
+    if (reservationUnit?.pk == null || begin == null || end == null) {
+      return;
+    }
+    const input: ReservationCreateMutationInput = {
+      begin: begin.toISOString(),
+      end: end.toISOString(),
+      reservationUnitPks: [reservationUnit.pk],
     };
-
+    createReservation(input);
+  };
+  const focusSlot: FocusTimeSlot = useMemo(() => {
+    const start = new Date(focusDate);
+    const [hours, minutes] = timeValue.split(":").map(Number);
+    start.setHours(hours, minutes, 0, 0);
+    const end = addMinutes(start, durationValue);
+    return {
+      start,
+      end,
+      isReservable: isSlotReservable(
+        start,
+        end,
+        false,
+        reservationUnit,
+        activeApplicationRounds
+      ),
+      durationMinutes: durationValue,
+    };
+  }, [
+    focusDate,
+    timeValue,
+    durationValue,
+    reservationUnit,
+    activeApplicationRounds,
+    isSlotReservable,
+  ]);
+  const durationOptions = useMemo(() => {
+    const {
+      minReservationDuration,
+      maxReservationDuration,
+      reservationStartInterval,
+    } = reservationUnit || {};
     if (
-      storedReservation?.reservationUnitPk === reservationUnit?.pk &&
-      storedReservation?.begin &&
-      storedReservation?.end
+      minReservationDuration == null ||
+      maxReservationDuration == null ||
+      reservationStartInterval == null
     ) {
-      setFocusDate(new Date(storedReservation.begin));
-      scrollToCalendar();
-      setInitialReservation(storedReservation);
-      removeStoredReservation();
-    } else if (hash === "calendar" && initialReservation) {
-      scrollToCalendar();
+      return [];
     }
+    return getDurationOptions(
+      minReservationDuration ?? undefined,
+      maxReservationDuration ?? undefined,
+      reservationStartInterval ?? 0,
+      t
+    );
+  }, [reservationUnit, t]);
 
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [initialReservation]);
+  const startingTimeOptions = useMemo(() => {
+    return getPossibleTimesForDay(
+      reservableTimeSpans,
+      reservationUnit?.reservationStartInterval,
+      focusDate,
+      reservationUnit,
+      activeApplicationRounds,
+      durationValue
+    );
+  }, [
+    reservableTimeSpans,
+    reservationUnit?.reservationStartInterval,
+    now,
+    focusDate,
+    durationValue,
+    isSlotReservable,
+  ]);
```

Comment:
This list look wrong. Why doesn't it match the params passed to `startingTimeOptions`